docs/react-v9/contributing/rfcs/react-components/convergence/react-18-upgrade.md

```diff
@@ -0,0 +1,225 @@
+
+The breaking changes would be the same as described in the beginning of this RFC.
+
+###### Pros
+
+- 👍 Less breaking changes for partners
+- 👍 React 18 support would land quicker
+
+##### Option 1-b: New Slot APIs
+
+If we choose to release a new major version, we could also take the opportunity to introduce new Slot APIs. This would be a good moment to overhaul the APIs to be more aligned with the future of Fluent UI, adding support for React 18 and beyond and making the slot API more flexible and easier to maintain/upgrade.
+
+The implementation APIs would be up to debate, but the idea is to make the APIs more flexible and easier to use, possibly decoupling from "@types/react" and making it easier to document and maintain. There are a few experiments in this direction, but nothing concrete yet.
+
+###### Pros
+
+- 👍 New Slot APIs would improve many aspects for Fluent:
+  - 👍 Easier types and possibly decoupled from "@types/react"
+  - 👍 Better prop tables
+  - 👍 Better documentation
+  - 👍 Future proof
+- 👍 We could also take the opportunity to fix some of the current issues with the slot API
+
+###### Cons
+
+- 👎 Time consuming and would delay the release
+- 👎 This would require more work to implement, test and release
+- 👎 This would be a big breaking change for partners, and it'll require good documentation and assistance
+
+> [!NOTE]
+> Due to how urgent React 18 upgrade is at the moment of the writing of this RFC, this option might not be viable, but it is written here anyway.
+
+##### Option 1-c: New Slot APIs with compat layer
+
+This is a variation of the previous option. We would introduce the new Slot APIs, but we could also add a compatibility layer to make the transition easier for partners. This would allow partners to upgrade to the new Slot APIs at their own pace, by using a compat package that would be a bridge between the old and new APIs.
+
+###### Pros
+
+- 👍 Partners could upgrade at their own pace
+- 👍 Same other pros as option 1-a
+
+###### Cons
+
+- 👎 It would require some strategy to deprecate this compat package in the future
+- 👎 Could bring some confusion to partners on why this is necessary
+- 👎 Same other cons as option 1-a
+
+> [!NOTE]
+> Due to how urgent React 18 upgrade is at the moment of the writing of this RFC, this option might not be viable, but it is written here anyway.
+
+###### Cons
+
+- 👎 In case we opt to develop new APIs for Slots in the future, it'd require a new breaking change
+
+### Option 2: Release a new minor version of Fluent UI
+
+This option involves releasing a new **minor** version of Fluent UI. This will break semantic versioning, but it will be a more seamless transition for partners.
+
+The release will include the necessary type changes, documentation updates and any other changes required.
+
+#### Pros
+
+- 👍 Avoid the fear of a new major release
```

Comment:
as far as I understand this would have same implications for consumers as having a new major version. the challenge is not with communication, the challenge is with shipping two incompatible APIs in one app together